rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/StateDedupedJobSchedulerDAO.java

```diff
@@ -19,25 +19,35 @@
      * When scheduling a job make sure its not already in the ready queue. A job
      * can be scheduled regardless of whether its in the running queue.
      * @param readyQueue the name of the ready queue
+     * @param runningQueue the name of the running queue
+     * @param readyAndRunningQueue the name of the ready and running queue
      * @param pausedTube the name of the paused tube
      * @param job the string representation of a job to schedule
      * @param runInMillis the timestamp in milliseconds when the job should be run.
      * @return 1 if job was scheduled or 0 otherwise.
      */
     @Query(
-        "local readyJobScore = redis.call('ZSCORE', $readyQueue$, $jobStr$)\n" +
+        "local readyJobScore = redis.call('ZSCORE', $readyQueue$, $job$)\n" +
+        "local runningJobScore = redis.call('ZSCORE', $runningQueue$, $job$)\n" +
+        "local readyAndRunningJobScore = redis.call('ZSCORE', $readyAndRunningQueue$, $job$)\n" +
         "local isPaused = redis.call('GET', $pausedTube$) \n" +
-        "if not readyJobScore and not isPaused then\n" +
-        "   redis.call('ZADD', $readyQueue$, $runInMillis$, $jobStr$)\n" +
+        "if not readyJobScore and not readyAndRunningJobScore and not isPaused then\n" +
+        "   if not runningJobScore then\n" +
+        "     redis.call('ZADD', $readyQueue$, $runInMillis$, $job$)\n" +
+        "   else\n" +
+        "     redis.call('ZADD', $readyAndRunningQueue$, $runInMillis$, $job$)\n" +
```

Comment:
so this mean's the third queue is actually for jobs that are currently running and also being put in ready queue?